contracts/bond/CollateralWhitelist.sol

```diff
@@ -3,38 +3,40 @@ pragma solidity ^0.8.0;
 
 import "@openzeppelin/contracts-upgradeable/token/ERC20/extensions/IERC20MetadataUpgradeable.sol";
 import "@openzeppelin/contracts-upgradeable/proxy/utils/Initializable.sol";
+import "@openzeppelin/contracts-upgradeable/utils/structs/EnumerableSetUpgradeable.sol";
+
+import "hardhat/console.sol";
 
 /**
  * @title Whitelist for collateral tokens.
  *
  * @notice Encapsulation of a ERC20 collateral tokens whitelist, indexed by their symbol.
  */
 abstract contract CollateralWhitelist is Initializable {
+    using EnumerableSetUpgradeable for EnumerableSetUpgradeable.AddressSet;
+
+    EnumerableSetUpgradeable.AddressSet private _whitelist;
+
     // Token symbols to ERC20 Token contract addresses
-    mapping(string => address) private _whitelist;
+    mapping(address => string) private _symbols;
 
     /**
-     * @notice The whitelisted ERC20 token address associated for a symbol.
-     *
-     * @return When present in the whitelist, the token address, otherwise address zero.
+     * @notice Whether the symbol has been whitelisted.
      */
-    function whitelistedCollateralAddress(string calldata symbol)
-        external
-        view
-        returns (address)
-    {
-        return _whitelist[symbol];
+    function isCollateralWhitelisted(address erc20) public view returns (bool) {
+        return _whitelist.contains(erc20);
     }
 
     /**
-     * @notice Whether the symbol has been whitelisted.
+     * @notice Returns a list of the whitelisted tokens' symbols
      */
-    function isCollateralWhitelisted(string memory symbol)
-        public
-        view
-        returns (bool)
-    {
-        return _whitelist[symbol] != address(0);
+    function getWhitelistSymbols() public view returns (string[] memory) {
```

Comment:
nit: `whitelistSymbols()` would be consistent with the other function getters.

Now this is a code style thing, for whatever reason I felt like dropping the `get` prefix from the getters and keeping the `set` prefix for the setters.
Probably to do with the TypeScript & Typechain combo not dealing well with function overloading (as is the JavaScript way 🙄 ).